File: Adm/Models/src/Itens_roupas.java
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.io.FileReader;
import java.io.FileWriter;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class Itens_roupas extends CRUD<Item_roupa> {

    @Override
    public void salvar() {
        try {
            FileWriter writer = new FileWriter("Data/item_roupa.json");
            Gson gson = new Gson();
            gson.toJson(objetos, writer); // Serializa os objetos e escreve no arquivo
            writer.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void abrir() {
        // Inicializa a lista objetos para evitar NullPointerException
        if (objetos == null) {
            objetos = new ArrayList<>();  // Se objetos for null, inicializa como lista vazia
        }

        try {
            FileReader reader = new FileReader("Data/item_roupa.json");
            Type listType = new TypeToken<List<Item_roupa>>(){}.getType();
            // Lê o arquivo JSON e desserializa para a lista de objetos
            List<Item_roupa> listaroupas_comb = new Gson().fromJson(reader, listType);
            if (listaroupas_comb != null) {
                objetos = listaroupas_comb; // Se o arquivo contiver dados, atualiza a lista objetos
            }
            reader.close();
        } catch (FileNotFoundException e) {
            // Arquivo não encontrado, não há dados para carregar
            objetos = new ArrayList<>(); // Inicializa a lista vazia
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Adm/Models/Itens_roupas.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;

namespace Adm.Models
{
    public class Itens_roupas : CRUD<Item_roupa>
    {

        public override void Salvar()
        {
            if (!Directory.Exists("../../Data"))
            {
                Directory.CreateDirectory("../../Data");
            }

            try
            {
                using (StreamWriter arquivo = new StreamWriter("../../Data/item_roupa.json"))
                {
                    var dados = new List<Dictionary<string, object>>();
                    foreach (var item_roupa in objetos)
                    {
                        dados.Add(item_roupa.ToDict());  
                    }

                    string json = JsonConvert.SerializeObject(dados);
                    arquivo.Write(json);
                }
            }
            catch (IOException e)
            {
                Console.WriteLine(e.Message);
            }
        }

        public override void Abrir()
        {
            try
            {
                using (StreamReader reader = new StreamReader("../../Data/item_roupa.json"))
                {
                    string json = reader.ReadToEnd();  // Lê o conteúdo do arquivo JSON
                    var listaItens_roupas = JsonConvert.DeserializeObject<List<Dictionary<string, object>>>(json);

                    if (listaItens_roupas != null)
                    {
                        objetos.Clear();
                        foreach (var obj in listaItens_roupas)
                        {
                            // Cria o objeto item_roupa a partir do dicionário
                            var item_roupa = new Item_roupa(
                                Convert.ToInt32(obj["id"]),
                                JsonConvert.DeserializeObject<List<int>>(obj["lista_id_roupas"].ToString())
                            );

                            objetos.Add(item_roupa);  // Adiciona o item_roupa à lista
                        }
                    }
                }
            }
            catch (FileNotFoundException)
            {
                // Não faz nada se o arquivo não for encontrado
            }
            catch (IOException e)
            {
        
            }
        }
    }
}
